dev/TEAL_opcodes.md

```diff
@@ -779,6 +754,55 @@ When A is a uint64, index 0 is the least significant bit. Setting bit 3 to 1 on
 - A uint64 formed from a range of big-endian bytes from A starting at B up to but not including B+8. If B+8 is larger than the array length, the program fails
 - Availability: v5
 
+## replace2 s
+
+- Opcode: 0x5c {uint8 start position}
+- Stack: ..., A: []byte, B: []byte &rarr; ..., []byte
+- Copy of A with the bytes starting at S replaced by the bytes of B. Fails if S+len(B) exceeds len(A)
+- Availability: v7
+
+## replace3
+
+- Opcode: 0x5d
+- Stack: ..., A: []byte, B: uint64, C: []byte &rarr; ..., []byte
+- Copy of A with the bytes starting at B replaced by the bytes of C. Fails if B+len(C) exceeds len(A)
+- Availability: v7
+
+## base64_decode e
+
+- Opcode: 0x5e {uint8 encoding index}
+- Stack: ..., A: []byte &rarr; ..., []byte
+- Decode Base-64 encoded byte-array A. Fail if A is not properly encoded.
+- **Cost**: 1 + 1 per 16 bytes of A
+- Availability: v7
+
+`base64` Encodings:
+
+| Index | Name | Notes |
+| - | ------ | --------- |
+| 0 | URLEncoding |  |
+| 1 | StdEncoding |  |
+
+
+A must be encoded as specified by <a href="https://rfc-editor.org/rfc/rfc4648.html">RFC 4648</a>.  Padding (`=`) characters may be present as specified by the RFC, or omitted entirely.  The encoding must follow the 'canonical' rules of section 3.5. `\n` and `\r` are allowed but completely ignored. No bytes other than those of the encoding alphabet, `=`, `\r`, and `\n` may appear.
```

Comment:
Why do you allow `\r` and `\n`?
Are they allowed anywhere or just at the beginning and end?

Reply:
They are allowed anywhere: "`\n` and `\r` are allowed but completely ignored".  It seems to be common to allow such, presumably from legacy uses of b64 like putting them in an email message with limited line length.  This is something the RFC says some implementation do (the Go implementation does), not something we made up out of thin air.